bionic/task_state.py

```diff
@@ -317,3 +329,54 @@ def _check_accessors_for_version_problems(self):
 
         for accessor in accessors_needing_saving:
             accessor.update_provenance()
+
+    # NOTE: This can be optimized further.
+    def new_state_for_completion(self, new_task_states_by_key, to_be_computed=True):
```

Comment:
It seems to me that the important thing about this method is not that it's *for completion* -- that's something that happens to every state regardless of the execution mode -- but rather that it *strips down* the state object in a way that allows it to be *transmitted* to *another process*. Maybe the name should reflect that?